modules/ballerina-logging/src/main/java/org/ballerinalang/logging/handlers/BallerinaLogFileHandler.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.logging.handlers;
+
+import java.io.IOException;
+import java.util.logging.FileHandler;
+
+/**
+ * A custom file handler for handling Ballerina user level log file
+ *
+ * @since 0.89
+ */
+public class BallerinaLogFileHandler extends FileHandler {
```

Comment:
This is a limitation of Java util logging. This is required to separate the logs to separate files. Since FileHandler can only be used once in the logging.properties file, for each additional log file we need, we have to extend the FileHandler class.